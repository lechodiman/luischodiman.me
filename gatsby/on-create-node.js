'use strict';

const path = require('path');
const _ = require('lodash');
const { createFilePath } = require('gatsby-source-filesystem');

const onCreateNode = ({ node, actions, getNode }) => {
  const { createNodeField } = actions;

  if (node.internal.type === 'MarkdownRemark') {
    if (typeof node.frontmatter.slug !== 'undefined') {
      const dirname = getNode(node.parent).relativeDirectory;
      createNodeField({
        node,
        name: 'slug',
        value: `/${dirname}/${node.frontmatter.slug}`,
      });

      const editLink = path
        .normalize(node.fileAbsolutePath)
        .replace(path.join(__dirname, '..'), '')
        .replace(/\\/gi, '/');

      createNodeField({
        node,
        name: 'editLink',
        value: `https://github.com/lechodiman/luischodiman.me/edit/master${editLink}`,
      });
    } else {
      const value = createFilePath({ node, getNode });
      createNodeField({
        node,
        name: 'slug',
        value,
      });
    }

    if (node.frontmatter.tags) {
      const tagSlugs = node.frontmatter.tags.map(
        tag => `/tag/${_.kebabCase(tag)}/`
      );
      createNodeField({ node, name: 'tagSlugs', value: tagSlugs });
    }

    if (node.frontmatter.category) {
      const categorySlug = `/category/${_.kebabCase(
        node.frontmatter.category
      )}/`;
      createNodeField({ node, name: 'categorySlug', value: categorySlug });
    }
  }
};

module.exports = onCreateNode;
