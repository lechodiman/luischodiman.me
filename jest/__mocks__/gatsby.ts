import React from 'react';
import gatsby from 'gatsby';

export default {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest
    .fn()
    .mockImplementation(
      ({
        activeClassName,
        activeStyle,
        getProps,
        innerRef,
        ref,
        replace,
        to,
        ...rest
      }) =>
        React.createElement('a', {
          ...rest,
          href: to
        })
    ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
