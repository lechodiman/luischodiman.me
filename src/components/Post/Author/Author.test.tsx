import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Author from './Author';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';

const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe('Author', () => {
  beforeEach(() => {
    mockedUseStaticQuery.mockReturnValue(siteMetadata);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Author />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
