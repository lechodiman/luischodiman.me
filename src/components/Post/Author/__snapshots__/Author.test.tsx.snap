// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div
  className="author"
>
  <p
    className="author__bio"
  >
    Test bio
    <a
      className="author__bio-twitter"
      href="https://www.linkedin.com/in/undefined"
      rel="noopener noreferrer"
      target="_blank"
    >
      <strong>
        Test name
      </strong>
       en LinkedIn
    </a>
  </p>
</div>
`;
